test/WebJobs.Script.Tests/FileMonitoringServiceTests.cs

```diff
@@ -27,6 +27,63 @@ public static void GetRelativeDirectory_ReturnsExpectedDirectoryName(string path
             Assert.Equal(expected, FileMonitoringService.GetRelativeDirectory(path, @"C:\Functions\Scripts"));
         }
 
+        [Theory]
+        [InlineData(@"NonExistentPath")]
+        [InlineData(null)]
+        public void InitializesEmptyOrMissingDirectorySnapshot(string rootScriptPath)
+        {
+            using (var directory = new TempDirectory())
+            {
+                // Setup
+                string tempDir = directory.Path;
+                Directory.CreateDirectory(Path.Combine(tempDir, "Host"));
+                File.Create(Path.Combine(tempDir, "host.json"));
+
+                var jobHostOptions = new ScriptJobHostOptions
+                {
+                    RootLogPath = tempDir,
+                    RootScriptPath = rootScriptPath,
+                    FileWatchingEnabled = true
+                };
+                var loggerFactory = new LoggerFactory();
+                var mockWebHostEnvironment = new Mock<IScriptJobHostEnvironment>(MockBehavior.Loose);
+                var mockEventManager = new ScriptEventManager();
+
+                // Act
+                FileMonitoringService fileMonitoringService = new FileMonitoringService(new OptionsWrapper<ScriptJobHostOptions>(jobHostOptions),
+                    loggerFactory, mockEventManager, mockWebHostEnvironment.Object);
+                Assert.False(fileMonitoringService.GetDirectorySnapshot().IsDefault);
+                Assert.True(fileMonitoringService.GetDirectorySnapshot().IsEmpty);
+            }
+        }
+
+        [Fact]
+        public void InitializesGetDirectorySnapshot()
+        {
+            using (var directory = new TempDirectory())
+            {
```

Comment:
Most of this method is identical to what you have above. Can we consolidate this logic to avoid duplication?